lib/engine/step/g_18_cz/sell_company_and_special_track.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require_relative '../special_track'
+
+module Engine
+  module Step
+    module G18CZ
+      class SellCompanyAndSpecialTrack < SpecialTrack
+        def actions(entity)
+          actions = []
+          abilities = tile_lay_abilities(entity)
+          actions << 'lay_tile' if abilities
+          actions << 'sell_company' if entity.company? && entity.owner == current_entity || entity == current_entity
+
+          actions
+        end
+
+        def process_sell_company(action)
+          corporation = action.entity
+          company = action.company
+          price = action.price
+
+          @game.bank.spend(price, corporation)
+
+          @log << "#{corporation.name} sells #{company.name} for #{@game.format_currency(price)} to the bank"
+
+          company.close!
+
+          @log << "#{company.name} closes"
+        end
+
+        def process_lay_tile(action)
+          spender = if !action.entity.owner
+                      nil
+                    elsif action.entity.owner.corporation?
+                      action.entity.owner
+                    else
+                      @game.current_entity
+                    end
+          lay_tile(action, spender: spender)
+
+          @game.skip_default_track unless @game.purple_tile?(action.tile)
+
+          abilities = @game.abilities(action.entity, :tile_lay, time: 'any')
+          abilities.each(&:use!)
+        end
+
+        def hex_neighbors(entity, hex)
+          return unless (abilities = tile_lay_abilities(entity))
+          return if abilities.all? { |ability| ability.hexes&.any? && !ability.hexes&.include?(hex.id) }
```

Comment:
any? -> !empty

Reply:
;) . I'm so used to c# linq any, I will make that mistake again and again